### ISSUE TITLE
Deprecation warnings with Mongo adapter (Node v10.11)

### ISSUE THREAD

**Issue by Nes-si** [parse-community/parse-server#5119]
Hi there.
I've got several deprecation warnings from Node:
~~~~
(node:12778) DeprecationWarning: collection.ensureIndex is deprecated. Use createIndexes instead. 
at Promise (/Volumes/data_hfs/Code/1-projects/Chisel/chisel-parse-server/node_modules/parse-server/lib/Adapters/Storage/Mongo/MongoCollection.js:98:29)`
~~~~
~~~~
(node:12778) DeprecationWarning: collection.findAndModify is deprecated. Use findOneAndUpdate, findOneAndReplace or findOneAndDelete instead.
at _adaptiveCollection.then.collection (/Volumes/data_hfs/Code/1-projects/Chisel/chisel-parse-server/node_modules/parse-server/lib/Adapters/Storage/Mongo/MongoStorageAdapter.js:418:95)`
~~~~
~~~~
(node:12778) DeprecationWarning: collection.count is deprecated, and will be removed in a future version. Use collection.countDocuments or collection.estimatedDocumentCount instead
at MongoCollection.count (/Volumes/data_hfs/Code/1-projects/Chisel/chisel-parse-server/node_modules/parse-server/lib/Adapters/Storage/Mongo/MongoCollection.js:60:50)
~~~~
~~~~
(node:12778) DeprecationWarning: collection.findAndRemove is deprecated. Use findOneAndDelete instead.
at MongoSchemaCollection.findAndDeleteSchema (/Volumes/data_hfs/Code/1-projects/Chisel/chisel-parse-server/node_modules/parse-server/lib/Adapters/Storage/Mongo/MongoSchemaCollection.js:171:46)
~~~~

### Environment Setup

- **Server**
  - parse-server version: 3.0.0
  - Operating System: macOS
  - Localhost

- **Database**
  - MongoDB version: 4.0.2
  - Localhost

**Comment by acinader** [comment]
@Nes-si thanks.  This issue was fixed by: #5025

which will be in the next release.

**Comment by sunshineo** [comment]
This error just happened to me. It is logged from Parse
```
(node:29) DeprecationWarning: collection.findAndRemove is deprecated. Use findOneAndDelete instead.
```
